recipes/server.rb

```diff
@@ -117,11 +117,18 @@ def remove_job(job, servernode)
 end
 clientnodes = clientsearchnodes.values
 
-# Filter out clients not in our environment, if applicable.
+# Filter out clients in the wrong environments, if applicable.
 if servernode['rdiff-backup']['server']['restrict-to-own-environment']
+  filterenvs = servernode['chef_environment']
```

Comment:
`.empty?` actually works for strings too.  It was in fact a mistake, but it just so happened to be valid.  I'll change it anyway though since it is not obvious.